### PR TITLE
fix astro dev init -v

### PR DESCRIPTION
**Pull request by andriisoldatenko**
resolves https://github.com/astronomer/issues/issues/2990

### CODE REVIEW

**Comment by andriisoldatenko** [comment]
practically LGTM:
@paolaperaza wdyt?
```
astro-cli dev init -v 0.0.0
Error: Unsupported Airflow Version specified. Please choose from: 2.1.0, 2.0.2, 2.0.0, 1.10.15, 1.10.14, 1.10.12, 1.10.10, 1.10.7, 1.10.5

Usage:
  astro dev init [flags]

Flags:
  -v, --airflow-version string   Version of airflow you want to deploy
  -h, --help                     help for init
  -n, --name string              Name of airflow project

Global Flags:
      --skip-version-check   skip version compatibility check

$ astro-cli dev init -v 2.0.2
Initialized empty astronomer project in /Users/andrii/workspace/src/github.com/astronomer/t_20210610_182938
$ cat Dockerfile
FROM quay.io/astronomer/ap-airflow:2.0.2-buster-onbuild%
```

few more examples when user trying to use `-v` while is not authenticated to a cluster:
```
astro-cli dev init -v 2.0.2
Error: you can't use --airflow-version option: No context set, have you authenticated to a cluster?
Usage:
  astro dev init [flags]

Flags:
  -v, --airflow-version string   Version of airflow you want to deploy
  -h, --help                     help for init
  -n, --name string              Name of airflow project

Global Flags:
      --skip-version-check   skip version compatibility check

```

and just fallback to default one:

```
astro-cli dev init
Initializing Airflow project
Not connected to Astronomer, pulling Airflow development files from 2.0.0-buster-onbuild
Initialized empty astronomer project in /Users/andrii/workspace/src/github.com/astronomer/t_20210610_184828
```

**Comment by adam2k** [comment]
@andriisoldatenko can you fix the merge conflicts for this?

**Review by adam2k — COMMENTED** [review]
@andriisoldatenko can you add tests to cover the new logic?

**Comment by andriisoldatenko** [comment]
> @andriisoldatenko can you add tests to cover the new logic?

yeah sure, i'm working on it right now.

**Comment by andriisoldatenko** [comment]
@adam2k PTAL!

File: cmd/airflow.go [hunk 1]
```diff
@@ -276,17 +276,9 @@ func airflowInit(cmd *cobra.Command, args []string, client *houston.Client, out
...
-	// if airflowVersion != "" && !acceptableVersion(airflowVersion, acceptableAirflowVersions) {
-	//  	return errors.Errorf(messages.ERROR_INVALID_AIRFLOW_VERSION, strings.Join(acceptableAirflowVersions, ", "))
-	// }
-
-	if len(defaultImageTag) == 0 {
-		defaultImageTag = "2.0.0-buster-onbuild"
-		fmt.Printf("Initializing Airflow project\nNot connected to Astronomer, pulling Airflow development files from %s\n", defaultImageTag)
+	defaultImageTag, err := prepareDefaultAirflowImageTag(airflowVersion, httpClient, client, out)
+	if err != nil {
+		return err
```
Comment (bote795) [inline cmd/airflow.go:281]: can we cover this case

File: cmd/airflow.go [hunk 2]
```diff
@@ -309,7 +301,7 @@ func airflowInit(cmd *cobra.Command, args []string, client *houston.Client, out
 	cmd.SilenceUsage = true
 
 	// Execute method
-	err := airflow.Init(config.WorkingPath, defaultImageTag)
+	err = airflow.Init(config.WorkingPath, defaultImageTag)
```
Comment (bote795) [inline cmd/airflow.go:304]: this line as well

File: cmd/utils.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		if airflowVersion != "" && !acceptableVersion(airflowVersion, acceptableAirflowVersions) {
+			return "", errors.Errorf(messages.ERROR_INVALID_AIRFLOW_VERSION, strings.Join(acceptableAirflowVersions, ", "))
+		}
+		if airflowVersion == "" {
+			defaultImageTag = ""
+		} else {
+			defaultImageTag = fmt.Sprintf("%s-buster-onbuild", airflowVersion)
+		}
+	} else if airflowVersion != "" {
+		return "", errors.New("you can't use --airflow-version option while not authenticated to a cluster")
```
Comment (aliotta) [inline cmd/utils.go:34]: aren't there several reasons why r.DoWithClient(houstonClient) could error? Its not always going to be b/c of not being authed

**Review by aliotta — APPROVED** [review]

**Review by bote795 — COMMENTED** [review]
lgtm, tests were added for new logic

**Pull request merged**